crates/solver/src/solver.rs

```diff
@@ -306,39 +307,44 @@ pub fn create(
     let http_instance_with_all_orders = http_solver::InstanceCache::default();
 
     // Helper function to create http solver instances.
-    let create_http_solver = |account: Account,
-                              url: Url,
-                              name: String,
-                              config: SolverConfig,
-                              filter_non_fee_connected_orders: bool,
-                              slippage_calculator: SlippageCalculator|
-     -> HttpSolver {
-        HttpSolver::new(
-            DefaultHttpSolverApi {
-                name,
-                network_name: network_id.clone(),
-                chain_id,
-                base: url,
-                client: http_factory.create(),
-                config,
-            },
-            account,
-            native_token,
-            token_info_fetcher.clone(),
-            buffer_retriever.clone(),
-            allowance_mananger.clone(),
-            order_converter.clone(),
-            if filter_non_fee_connected_orders {
-                http_instance_with_filtered_orders.clone()
-            } else {
-                http_instance_with_all_orders.clone()
-            },
-            filter_non_fee_connected_orders,
-            slippage_calculator,
-            market_makable_token_list.clone(),
-            *domain,
-        )
-    };
+    let create_http_solver =
+        |account: Account,
+         url: Url,
+         name: String,
+         config: SolverConfig,
+         filter_non_fee_connected_orders: bool,
+         slippage_calculator: SlippageCalculator,
+         instance_uploader_config: Option<crate::s3_instance_upload::Config>|
+         -> HttpSolver {
+            HttpSolver::new(
+                DefaultHttpSolverApi {
+                    name,
+                    network_name: network_id.clone(),
+                    chain_id,
+                    base: url,
+                    client: http_factory.create(),
+                    config,
+                },
+                account,
+                native_token,
+                token_info_fetcher.clone(),
+                buffer_retriever.clone(),
+                allowance_mananger.clone(),
+                order_converter.clone(),
+                if filter_non_fee_connected_orders {
+                    http_instance_with_filtered_orders.clone()
+                } else {
+                    http_instance_with_all_orders.clone()
+                },
+                filter_non_fee_connected_orders,
+                slippage_calculator,
+                market_makable_token_list.clone(),
+                *domain,
+                instance_uploader_config
+                    .map(crate::s3_instance_upload::S3InstanceUploader::new)
+                    .map(Arc::new),
```

Comment:
I guess this could be moved out of `create_http_solver` such that we can only ever have a single instance of `S3InstanceUploader`. I know we are currently only creating a single instance but in a slightly weird way.

Reply:
The way it is implemented now, if we wanted multiple uploaders then they would need to be configured separately. But I can still pass in the uploader here directly instead of the config.